### ISSUE TITLE
Bump environment python version to 3.10

### ISSUE THREAD

**Issue by ConstensouxAlexis** [Sanofi-Public/enact-pipeline#14]
Hello,
Thank you for sharing this tool. I think that the enact conda env should be created with python10, because run_enact raises an error for python version below 3.10 (I think that operator | was added in python3.10):

`  File "../analysis_workspaces/visium_hd_telocytes/code/envs/enact_py_env/lib/python3.9/site-packages/numpyro/diagnostics.py", line 233, in <module>
    samples: dict | np.ndarray, prob: float = 0.90, group_by_chain: bool = True
TypeError: unsupported operand type(s) for |: 'type' and 'type'`

**Comment by AlbertPlaPlanas** [comment]
You are right about operator '|', nonetheless I do not recall having seen this error with enact when using python 3.9.

Can you paste the full stack trace to see in which line(s) of our code this is happening?

On the meantime, you can test it yourself locally by upgrading the python version in the following two files:
- https://github.com/Sanofi-Public/enact-pipeline/blob/c1282932af1f73e825a5afdf4ca5a27ddc64a197/setup_py_env.sh#L11
- https://github.com/Sanofi-Public/enact-pipeline/blob/c1282932af1f73e825a5afdf4ca5a27ddc64a197/Makefile#L10
